Fix/flex images v1.8 (#647)
ZupIT/beagle-docs: content/pt/resources/components-positioning/overview.md

## **Introdução**

O Beagle usa uma dependência chamada **Yoga layout** para posicionar seus componentes na tela de uma aplicação. Para isso, o Yoga calcula a posição dos elementos e, em seguida, faz sua renderização.
![](/shared/flex/image125.png)

## Sobre Yoga

O [**Yoga**](https://yogalayout.com/) é um framework desenvolvido pelo Facebook que posiciona os elementos em uma tela usando o conceito Flexbox aplicados ao Android e iOS.

Flexbox é um conceito do CSS que é utilizado para posicionar elementos em contêineres dinamicamente para que, independentemente das dimensões de seu aplicativo, você possa manter um layout flexível.

> Seguem alguns conceitos-chave sobre o Flexbox:

Reviewer: ```suggestion
Veja abaixo alguns conceitos-chave sobre o Flexbox:
```

- **Main Axis:** 
- **Cross Axis:** 
- **Main Size:**

No caso do Main Size, é possível ter **3 tipos diferentes de dimensões:**

1. **Cross Size:** Tamanho da dimensão cross axis.
2. **Main Start and Main End:** Começo e o fim do main axis.
3. **Cross Start and Cross End:** Começo e o fim do cross axis.

**‌**Esses valores para os eixos(axis) dependem de como é definida a propriedade flex-direction.

Se esta propriedade for definida como:

| **row ou row-reverse:** | **column ou column-reverse:** |
| :-----------: | :----------------------------------------------------------: |
| O eixo principal (main axis) será horizontal e o eixo transversal (cross axis) será vertical. | O eixo principal (main axis) será vertical e o eixo transversal (cross axis) será horizontal. |
| {{< figure src="/shared/flex/row.png" width="150">}} | {{< figure src="/shared/flex/column.png" width="150">}} |

{{% alert color="info" %}}

> Important: O valor default do Beagle é `column` ou seja, os elementos serão dispostos em colunas por padrão.

Reviewer: ```suggestion
Important: O valor default do Beagle é `column`, ou seja, os elementos serão dispostos em colunas por padrão.
```

{{% /alert %}}

## Propriedades

> Existem algumas propriedades que o Yoga Layout pode usar para inserir, mover ou excluir componentes. A seguir temos algumas delas:

Reviewer: ```suggestion
O Yoga Layout pode usar alguams propriedades para inserir, mover ou excluir componentes. Veja a seguir algumas delas:
```

### **UnitValue**

> Os atributos **Base, Size, Margin, Padding e Position** recebem um valo(`UnitValue`) do tipo `Double` e um `UnitType`, do tipo `enum` com as seguintes opções:

Reviewer: ```suggestion
Os atributos **Base, Size, Margin, Padding e Position** recebem um valor (`UnitValue`) do tipo `Double` e um `UnitType` do tipo `enum`, ambos com as opções:
```

| **UnitType** | Definição                                                        |
| :----------- | :--------------------------------------------------------------- |
| **REAL**     | Aplica o valor `Double`                                          |
| **PERCENT**  | Aplica o valor `Double` em forma de percentual do tamanho do pai |
| **AUTO**     | Segue o valor do pai. Exceto quando possui tamanho próprio       |

{{% alert color="warning" %}}

> **Importante!** Os atributos acima quando referenciados em um projeto WEB tem como opção `default is` o `UnitType.AUTO`.

Reviewer: ```suggestion
 Quando os atributos acima são referenciados em um projeto WEB, a opção `default is` é o `UnitType.AUTO`.
```
